pkg/network/server_test.go

```diff
@@ -115,11 +117,43 @@ func TestServerStartAndShutdown(t *testing.T) {
 		s.register <- p
 
 		assert.True(t, s.services["fake"].(*fakeConsensus).started.Load())
+		require.True(t, s.started.Load())
 
 		s.Shutdown()
 
+		require.False(t, s.started.Load())
 		require.True(t, s.services["fake"].(*fakeConsensus).stopped.Load())
 	})
+	t.Run("double start", func(t *testing.T) {
+		s := newTestServer(t, ServerConfig{})
+		startWithCleanup(t, s)
+
+		// Ensure the server is started.
+		require.True(t, s.started.Load(), "server should be marked as started after initial Start call")
+
+		// Attempt to start the server again.
+		s.Start()
+
+		require.True(t, s.started.Load(), "server should still be marked as started after second Start call")
+	})
+	t.Run("double shutdown", func(t *testing.T) {
+		s := newTestServer(t, ServerConfig{})
+		go s.Start()
+		require.Eventually(t, func() bool {
+			return s.runLoopStarted.Load()
+		}, time.Second, 10*time.Millisecond)
+
+		// Ensure the server is started.
+		require.True(t, s.started.Load(), "server should be marked as started after initial Start call")
```

Comment:
Useless check, it's already done inside `require.Eventually(...)` five lines above.